### DISCUSSION TITLE
What are the usecases for this library?

### DISCUSSION THREAD

**Discussion by Evilweed** [niklasvh/html2canvas#2580] [Q&A]
Hey there! :) Just stumbled uppon this library and i am trying to figure out what are the usecases for it? I love technology and learning. Still i cannot think on the spot a usecase in which this is better than just performing regular screenshot.

**Comment by niklasvh** [comment]
I can't speak from personal experience, as I don't use it for anything. The main difference between this library and taking a screenshot manually, is that this library can perform it programmatically, on the behalf of the user. But based on where I've seen it used a lot are for example applications that want to convert html into a pdf document, capture some visual state from the screen (for example an app that allows you to move elements around etc, like paint), and then just general visualization apps like what [Google did with their Thanos snap](https://techcrunch.com/2019/04/26/thanos-will-snap-away-your-google-search-results/):

https://user-images.githubusercontent.com/551347/125599324-5a139e55-4211-4151-aded-157dc987cc02.mp4

**Comment by AgentSmith0** [comment]
This library is also useful for bug reporting or feedback systems. The user will be able to mark a problem directly without taking and uploading a screenshot.
